plugin/documents.py

```diff
@@ -635,6 +639,17 @@ def render_highlights_on_main_thread() -> None:
 
         sublime.set_timeout(render_highlights_on_main_thread)
 
+    # --- textDocument/semanticTokens ----------------------------------------------------------------------------------
+
+    def _do_semantic_tokens_async(self) -> None:
+        session = self.session_async("semanticTokensProvider")
+        if not session:
+            return
+        session_view = session.session_view_for_view_async(self.view)
+        if not session_view:
+            return
+        session_view.session_buffer.do_semantic_tokens_async()
```

Comment:
something like this should work

```suggestion
        sb = next(self.session_buffers_async())
        sb.do_semantic_tokens_async()
```

Reply:
I think that doesn't guarantee that it is the buffer of the active view, or am I wrong?

Reply:
It is, but there can be multiple because there can be multiple langservers attached to the view so it's wrong anyway to take the first one :)